Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ILLLagrange.py

```
@@ -0,0 +1,308 @@
+                        if "CNTS" in next_line:
+                            detector_counts_col = next_line.index("CNTS")
+                        else:
+                            detector_counts_col = self.DETECTOR_COUNT_COL_DEFAULT
+                            self.log().warning("Could not find detector counts column in file {}. "
+                                               "Defaulting to column {}".format(file, self.DETECTOR_COUNT_COL_DEFAULT))
+                        if "M1" in next_line:
+                            monitor_counts_col = next_line.index("M1")
+                        else:
+                            monitor_counts_col = self.MONITOR_COUNT_COL_DEFAULT
+                            self.log().warning("Could not find monitor counts column in file {}. "
+                                               "Defaulting to column {}".format(file, self.MONITOR_COUNT_COL_DEFAULT))
+                        break
+                f.close()
+
+            if header_index < 0:
+                self.log().warning("Could not determine header length in file {}. "
+                                   "Defaulting to {}".format(file, self.HEADER_SIZE_DEFAULT))
+                header_index = self.HEADER_SIZE_DEFAULT
+                energy_col = self.ENERGY_COL_DEFAULT
+                detector_counts_col = self.DETECTOR_COUNT_COL_DEFAULT
+                monitor_counts_col = self.MONITOR_COUNT_COL_DEFAULT
+
+            # load the relevant data, which position we just determined - angle, monitor count, detector count
+            data = np.loadtxt(file,
+                              dtype=float,
+                              skiprows=header_index + 2,
+                              usecols=(energy_col, monitor_counts_col, detector_counts_col))
+
+            loaded_data = np.vstack((loaded_data, data)) if loaded_data is not None else data
+
+        return loaded_data
```

Here, if one of the file cannot be read, this will be silently ignored. If all files cannot be read, an obscure error will come from the workspace creation line 100.

I would add a check on `data` to verify that something was actually read from the file and a final check on `loaded_data` as well.